src/app/workflow/view/view.component.html

```diff
@@ -40,3 +29,27 @@
     Delete
   </button>
 </ng-container>
+<button
+  type="button"
+  mat-button
+  color="accent"
+  class="btn-block"
+  *ngIf="!version.frozen && version.referenceType !== 'BRANCH'"
+  (click)="snapshotVersion()"
+  data-cy="dockstore-snapshot"
+  [matTooltip]="snapshotTooltip"
+>
+  Snapshot
+</button>
+<button
+  mat-button
+  id="DOIButton"
+  color="accent"
+  class="btn-block"
+  (click)="requestDOIForWorkflowVersion()"
+  [disabled]="isRefreshing$ | async"
```

Comment:
Haven't gotten to TS code yet, but probably should be disabled on requestingDOI, not refreshing.